### ISSUE TITLE
[Feature request] Custom Relay Servers and Relay Server List

### ISSUE THREAD

**Issue by Tippek** [libretro/RetroArch#8124]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

[Feature request] Add Relay Server List and add posibility send information about mitm-server  to lobby. Add posibility to RetroArch to use Relay Server List.

### Expected behavior

We need more Relay Servers.

### Actual behavior

There are only 2-3 Relay Servers actually.

### Steps to reproduce the bug

1. [First step]
2. [Second step]
3. [and so on...]

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [version/commit]

### Environment information

- OS: [The operating system you're running]
- Compiler: [In case you are running local builds]

**Comment by orbea** [comment]
Duplicate of issue https://github.com/libretro/RetroArch/issues/5732, but since you followed the issue template lets keep this one instead.

**Comment by bparker06** [comment]
Can you elaborate on exactly what you're asking? I'm having a very hard time understanding.

**Comment by orbea** [comment]
@bparker06 Some users want to be able to setup and use their own relay servers and not rely upon those set up by RetroArch devs.

**Comment by fr500** [comment]
They can setup a relay server for themselves just fine.
The only thing they can't do is add them to the lobby.

It could be done for sure, a relay would connect to the lobby to announce it exists, then the clients that

**Comment by gustavolinux** [comment]
> They can setup a relay server for themselves just fine.
> The only thing they can't do is add them to the lobby.
> 
> It could be done for sure, we could add something to the netplay announce indicating it's using a custom relay server, but the whole multiport coordination is more complex, it needs two way communication between the lobby server and the relay server.

How can I setup my private relay server? Is there a binary relay server out there?

**Comment by evacomaroski** [comment]
@gustavolinux would like to know as well.

+1 just chiming in on this!

**Comment by goldug** [comment]
Yes, I'm chiming in on this too. It would be great to set up my own custom relay server. When I don't use a relay server, the game lags a lot for some reason, even though I have a new and powerful one. I run a couple of private servers that have no real use at the moment, so I wanna set one up as a relay.

**Comment by Stixil** [comment]
I know this is a late response, but I wanted to +1 this issue. Custom relay server for end users to run on their own networks or private servers to host would be a dream come true.

**Comment by pashdown** [comment]
I'd be happy to setup a public relay server if given the opportunity.

**Comment by Stixil** [comment]
Checking on this. Is there still no binary server for users to host? Would give netplay far more mileage and bring new people to RetroArch!

**Comment by ditaccio** [comment]
Second this.
We could start our own relay servers, on AWS, for example

**Comment by zhy9934** [comment]
Is there a available relay server program？thinks

**Comment by KoNickss** [comment]
descentralization of relays would be a great thing to do in the spirit of foss software

**Comment by pashdown** [comment]
Is this ever going to be possible?

**Comment by jobimajobima** [comment]
> Is there a available relay server program？thinks

this https://github.com/libretro/netplay-tunnel-server
